src/Traits/ReflectionClassLikeTrait.php

```diff
@@ -401,46 +425,19 @@ function (array &$result, \ReflectionClass $instance, $isParent) {
     }
 
     /**
-     * Returns a bitfield of the access modifiers for this class.
-     *
-     * @link http://php.net/manual/en/reflectionclass.getmodifiers.php
-     *
-     * NB: this method is not fully compatible with original value because of hidden internal constants
-     *
-     * @return int
+     * {@inheritDoc}
      */
-    public function getModifiers()
+    public function getModifiers(): int
     {
-        $modifiers = 0;
-
-        if ($this->isFinal()) {
-            $modifiers += \ReflectionClass::IS_FINAL;
-        }
-
-        if (PHP_VERSION_ID < 70000 && $this->isTrait()) {
-            $modifiers += \ReflectionClass::IS_EXPLICIT_ABSTRACT;
-        }
-
-        if ($this->classLikeNode instanceof Class_ && $this->classLikeNode->isAbstract()) {
-            $modifiers += \ReflectionClass::IS_EXPLICIT_ABSTRACT;
-        }
-
-        if ($this->isInterface()) {
-            $abstractMethods = $this->getMethods();
-        } else {
-            $abstractMethods = $this->getMethods(\ReflectionMethod::IS_ABSTRACT);
-        }
-        if (!empty($abstractMethods)) {
-            $modifiers += \ReflectionClass::IS_IMPLICIT_ABSTRACT;
-        }
+        $this->initializeInternalReflection();
```

Comment:
It is pity that this method triggers now loading of internal core reflection, and AFAIK, this method was used somewhere in code-generation part too. This means it can be break logic of weaving data inside class (we use static analysis, because we run **before** composer autoloader, thus class can not be loaded here properly and can not be reflected)

Reply:
Re-added and fixed for PHP 8.0

https://github.com/goaop/parser-reflection/pull/122/commits/894cc827e0a0f14cc9426ab14cec2de37d67201c

Reply:
Works with PHP 8.1 also